CONTRIBUTING.md

```diff
@@ -0,0 +1,388 @@
+
+Here's the full code of the node:
+
+```py
+import numpy as np
+
+from nodes.impl.pil_utils import convert_to_BGRA
+from nodes.properties import expression
+from nodes.properties.inputs import ImageInput, SliderInput
+from nodes.properties.outputs import ImageOutput
+from nodes.utils.utils import get_h_w_c
+
+from .. import adjustments_group
+
+
+@adjustments_group.register(
+    schema_id="chainner:image:opacity",
+    name="Opacity",
+    description="Adjusts the opacity of an image. The higher the opacity value, the more opaque the image is.",
+    icon="MdOutlineOpacity",
+    inputs=[
+        ImageInput(),
+        SliderInput(
+            "Opacity",
+            maximum=100,
+            default=100,
+            precision=1,
+            controls_step=1,
+            unit="%",
+        ),
+    ],
+    outputs=[
+        ImageOutput(
+            image_type=expression.Image(size_as="Input0"),
+            channels=4,
+        )
+    ],
+)
+def opacity_node(img: np.ndarray, opacity: float) -> np.ndarray:
+    """Apply opacity adjustment to alpha channel"""
+
+    # Convert inputs
+    c = get_h_w_c(img)[2]
+    if opacity == 100 and c == 4:
+        return img
+    imgout = convert_to_BGRA(img, c)
+    opacity /= 100
+
+    imgout[:, :, 3] *= opacity
+
+    return imgout
+```
+
+We can see the code is roughly split into 2 parts: `@adjustments_group.register(...)` and `def opacity_node`.
+
+The first part defines [the node's metadata](#node-metadata). Nodes are registered for a specific node group (see [the section on categories](#categories)). In this case, the node is registered for the `adjustments_group` node group. Name, description, and icon should be self-explanatory. The `schema_id` is a unique identifier for the node. Input and outputs define rich metadata about the arguments and return value of the node. E.g. The `SliderInput` contains a lot more information than `opacity: float`. See [the section on inputs and outputs](#inputs-and-outputs) for more information.
+
+The second part is the actual node implementation. The implementation is rather straightforward. The node takes an image and an opacity value as input. Aside from handling a special case, it just multiplies the alpha channel of the image with the opacity value. Importantly, the input image is _not_ modified. Instead, the image is copied (`convert_to_BGRA` guarantees a copy) and the copy is modified and returned. See [the nodes section](#nodes) as to why copying is necessary.
+
+It should be noted that metadata is used to verify the correctness of the implementation. E.g. the type hints of `img` and `opacity` are checked against the input metadata. So you can rely upon the fact that `opacity` is a number between 0 and 100. It's similar for the return type annotation. E.g. in this example, `ImageOutput` will validate that the returned image has 4 channels (as declared by `channels=4`) and will throw an error otherwise.
```

Comment:
Because of this, we should probably mention that we require you to use type hints in your node function